### PR TITLE
Add option for prepending directories to the module search path

### PR DESCRIPTION
**Pull request by pdesaulniers**
Initial work on https://github.com/teal-language/tl/issues/92.

This PR adds the `-I --include` option for prepending directories to `package.path`. It also adds an `include` key in `tlconfig.lua`.

### CODE REVIEW

File: tl [hunk 1]
```diff
@@ -18,10 +18,21 @@ local function die(msg)
    os.exit(1)
 end
 
+local function find_in_sequence(seq, value)
```
Comment (hishamhm) [inline tl:21]: The PR looks great but the indentation is a bit off? (4 spaces in a few places)

**Comment by hishamhm** [comment]
I was thinking we could update `package.cpath` as well with `-I` (it could be useful for `tl run`) but that would be an "extra", I think functionality-wise this PR is good to go as is already!

File: tl (hunk 1 above)
Comment (pdesaulniers) [reply tl:21]: My bad! Should be fixed, I think.

**Comment by pdesaulniers** [comment]
> I was thinking we could update package.cpath as well with -I

OK, I don't mind adding this right now. I went with `package.path` first since it's more useful for my use-case.

**Comment by pdesaulniers** [comment]
It should now set `package.cpath` too.

File: spec/cli/include_spec.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      local pd = io.popen("./tl -I spec check " .. name, "r")
+      local output = pd:read("*a")
+      util.assert_popen_close(true, "exit", 0, pd:close())
+      assert.match("0 errors detected", output, 1, true)
+   end)
+   it("adds a directory to package.cpath", function()
+      local name = util.write_tmp_file(finally, "foo.lua", [[
+         print(package.cpath)
+      ]])
+      local pd = io.popen("LUA_CPATH=\"/usr/lib/lua/?.so\" ./tl run -I spec/cli/ " .. name, "r")
```
Comment (hishamhm) [inline spec/cli/include_spec.lua:21]: I think this test will fail if people have things like LUA_CPATH_5_3 set in their environments (I do :grin: ). I believe it's okay to just do `print(package.cpath:match("spec/cli/") ~= nil)` above and check for `true` below (should also keep it crossplatform enough!)

Comment (pdesaulniers) [reply spec/cli/include_spec.lua]: Good call! Done.

**Comment by hishamhm** [comment]
@pdesaulniers Merged! :grinning:

**Pull request merged**